src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncProgressResolver.cs

```diff
@@ -117,6 +124,25 @@ public long FindBestFullState()
                 }
             }
 
+            if (bestFullState == 0)
+            {
+                if (_bestStateCounter != -1)
+                {
+                    return _bestStateCounter;
+                }
+                byte[] bestStatePersistedCounter = _stateDb.Get(LatestBlockWithFullState);
+                Interlocked.Exchange(ref _bestStateCounter, bestStatePersistedCounter != null ? BitConverter.ToInt64(bestStatePersistedCounter) : 0);
+                return _bestStateCounter;
+            }
+            else
+            {
+                if (bestFullState != _bestStateCounter)
+                {
+                    _stateDb.Set(LatestBlockWithFullState, BitConverter.GetBytes(bestFullState));
+                    Interlocked.Exchange(ref _bestStateCounter, bestFullState);
+                }
+            }
+
```

Comment:
I don't get it why we are saving it now in db, but still always go through SearchForFullState.
If we want to have a db field maybe we should save it when we actually know we have state for block:
1. When we finish StateSync
2. When we process new block